content/blog/2024-01-16.md

```diff
@@ -0,0 +1,28 @@
++++
+title = "2024-01-16"
+date = 2024-01-16
++++
+
+---
+
+Today is Tuesday, day 016 of 366 (4.37%). We're in week 03 of 52 (5.76%).
+
+## Planning
+
+What I am going to do today:
+
+- [x] work on the [edit snippet page](https://github.com/OmnicodeSolutions/luisa_drf_flutter_client/blob/main/lib/edit_snippet.dart)
+
+## Activities executed
+
+Today I was able to fix the problem from yesterday. The issue was that I had rows that were too big inside columns that were trying to fit the rows. So, I rearranged the code. After fixing that, I added the [text inputs for the title and code](https://github.com/OmnicodeSolutions/luisa_drf_flutter_client/blob/f17d92b65b81a078b6de92deb6775a4f62bb56ec/lib/edit_snippet.dart#L93C25-L110C27), and I'm trying to add the linenos checkbox. However, I'm having trouble with it since the [Checkbox class](https://api.flutter.dev/flutter/material/Checkbox-class.html) doesn't have an initial value property, and there is no way to check if a late variable (which is how I initially thought about treating the linenos value) has been initialized. I'm trying to make the checkbox work, and I will search for other solutions tomorrow, but I'm leaning toward using another class, like a button.
+
+## Learnings
+
+Today I learned about the Checkbox class, whose name is pretty self-explanatory, and about the [RenderFlex overflow error](https://docs.flutter.dev/testing/common-errors#a-renderflex-overflowed), which is pretty common and mostly happens when there are too long items inside columns. The parent tries to accommodate it but ends up getting bigger than the screen. Fixing it can be as simple as wrapping the item with `Expanded()` or refactoring your code. The error looks something like this:
+
+```shell
+Another exception was thrown: RenderBox was not laid out:
+_RenderDecoration#9a7f2 relayoutBoundary=up10 NEEDS-PAINT
+NEEDS-COMPOSITING-BITS-UPDATE
+
```

Comment:
```suggestion
```shell
Another exception was thrown: RenderBox was not laid out:
_RenderDecoration#9a7f2 relayoutBoundary=up10 NEEDS-PAINT
NEEDS-COMPOSITING-BITS-UPDATE
```
```

Reply:
You forgot to close the ```